### PR TITLE
Implement User Deactivation and Reactivation Endpoints in AuthController

### LINKED ISSUE #5: Database: Set Up Database Infrastructure
Set up the initial database using MySQL or another chosen database system. Create the necessary configurations and environments for local development.

### LINKED ISSUE #17: Implement User Deactivation API
### Objective
Create an API endpoint for deactivating user accounts while preserving data.

### Tasks
- Develop an endpoint to deactivate a user account.
- Ensure that the deactivated user data is retained in the system.

### Acceptance Criteria
- API endpoints are implemented and documented.
- Proper error handling and validation are in place.

### Labels
- enhancement
- backend
- API

### LINKED ISSUE #29: Email Not Received After Registration
### Description
When a user registers, a message indicates that an email has been sent for verification. However, users do not receive any email in their inbox. This issue needs investigation to determine if it's related to the code implementation or the email configuration settings.

### Steps to Reproduce
1. Attempt to register a new user account.
2. Observe the message stating that the email has been sent.
3. Check the inbox for the email; it is not received.

### Expected Behavior
After registering, the user should receive a verification email in their inbox.

### Actual Behavior
The application indicates that an email has been sent, but no email is received.

### Potential Areas to Investigate
1. **Email Sending Code**: Review the implementation in the authentication service responsible for sending emails. Ensure it handles sending correctly and logs any errors that may occur during the process.
2. **Email Configuration**: Check the email service configuration (SMTP settings) to ensure it is set up correctly. Verify that credentials, ports, and host settings are accurate.
3. **Email Provider Issues**: There might be issues with the email provider or the sender's address. Ensure that emails are not being sent to spam or filtered by the email provider.
4. **Logging**: Implement logging around the email sending process to capture any errors or issues that occur when attempting to send emails.

### Additional Notes
- If applicable, provide details about the email service provider used (e.g., SMTP server, third-party email service).
- Mention any relevant logs or error messages observed during the registration process or while sending the email.

### Suggested Next Steps
- Review the email sending code and configuration.
- Test the email functionality with different email addresses.
- Verify whether the emails are being sent and received in other inboxes or domains.

### PR DESCRIPTION
**Pull request by AhmedFatthy1040**
### closes #17 

This PR introduces two new endpoints in the `AuthController` for managing user accounts: deactivating and reactivating users. The functionality allows administrators to manage user accounts effectively, enhancing user management capabilities.

#### Changes Made:
- Added the following methods in the `AuthService` interface:
  - `boolean deactivateUser(Long userId);`
  - `boolean reactivateUser(Long userId);`
- Implemented the above methods in the `AuthService` class.
- Added corresponding endpoints in the `AuthController`:
  - `POST /api/auth/{id}/deactivate` to deactivate a user.
  - `POST /api/auth/{id}/reactivate` to reactivate a user.
- Added error handling for user account status during login.

#### Screenshots:
1. **Deactivation Endpoint in Postman:**

   Screenshot from 2024-10-03 10-50-53

   
3. **Reactivation Endpoint in Postman:**

   Screenshot from 2024-10-03 10-51-08

   
5. **Error Handling during Login:**

   Screenshot from 2024-10-03 10-50-27

#### Testing:
- Successfully tested the deactivation and reactivation endpoints using Postman, confirming that the user account status updates correctly.
- Verified that the login process correctly handles disabled and locked accounts.

#### Checklist:
- [x] Code compiles without errors.

#### Also closes #29 as i fixed the package name problem in MailConfig.java.

### CODE REVIEW

File: src/main/java/com/activecourses/upwork/controller/auth/AuthController.java
Comment (AhmedMohamedAbdelaty) [inline src/main/java/com/activecourses/upwork/controller/auth/AuthController.java:1]: Can you make sure that only admins can use the `deactivateUser` and `reactivateUser` methods?

Comment (AhmedMohamedAbdelaty) [reply src/main/java/com/activecourses/upwork/controller/auth/AuthController.java:1]: You can use `@PreAuthorize` annotation.

Comment (AhmedFatthy1040) [reply src/main/java/com/activecourses/upwork/controller/auth/AuthController.java:1]: Done.

**Review by AhmedMohamedAbdelaty — APPROVED** [review]

**Review by AhmedYahia74 — APPROVED** [review]
LGTM

**Pull request merged**